src/services/utilities.ts

```diff
@@ -2808,6 +2809,125 @@ namespace ts {
         }
     }
 
+    export interface PackageJsonImportFilter {

```

Comment:
All this added stuff is just moved from completions.ts, importFixes.ts, or project.ts with very little change. The sharing between those three was ad-hoc and getting unwieldy.